### PR TITLE
Fix missing signals and restructure ( Fixes #79 )

### LINKED ISSUE #79: Missing Signals
There a few "No such signal" errors occurring when running the program.

### PR DESCRIPTION
**Pull request by jeffh1996**
**Signals** 
The missing signal errors in #79 should be fixed. 

**Structure**
The directory structure has also been changed to match the test tool structure.

**setMppt**
The setMppt is used to update the list of Mppt data and then send out a signal. I ran into the QList problem again when I was testing this part. It turns out that the QList.replace function was trying to access non existent data. This has been fixed in this PR by initializing the list of Mppts in the constructor (also tested in my test branch JsonTest, the Mppt values on the display dashboard do get updated and there is no QList error anymore).

### CODE REVIEW

**Comment by jeffh1996** [comment]
There were more missing signals with the latest merge.

**Comment by jeffh1996** [comment]
All missing signals have been fixed up to now including the 5 new ones in the latest PR.

File: src/ViewLayer/DebugDisplay/ControlPage/ControlView/ControlView.cpp [hunk 1]
```diff
@@ -31,15 +31,15 @@ void ControlView::connectDriverControls(DriverControlsPresenter& driverControlsP
     connect(&driverControlsPresenter, SIGNAL(aliveReceived(bool)),
             this, SLOT(aliveReceived(bool)));
 
-    connect(&driverControlsPresenter, SIGNAL(lowHeadlightsReceived(bool)),
+    connect(&driverControlsPresenter, SIGNAL(headlightsLowReceived(bool)), //
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/ControlPage/ControlView/ControlView.cpp]: Are the `//`'s still necessary?

File: src/ViewLayer/DisplayDashboard/DisplayDashboardView/DisplayDashboardView.cpp [hunk 2]
```diff
@@ -106,8 +106,8 @@ void DisplayDashboardView::connectMppt(MpptPresenter& mpptPresenter)
 {
     connect(&mpptPresenter, SIGNAL(mpptReceived(int, Mppt)),
             this, SLOT(mpptReceived(int, Mppt)));
-    connect(&mpptPresenter, SIGNAL(mpptPowerReceived(double)),
-            this, SLOT(mpptPowerReceived(double)));
+    //connect(&mpptPresenter, SIGNAL(mpptPowerReceived(double)),
```
Comment (bill-luu) [inline src/ViewLayer/DisplayDashboard/DisplayDashboardView/DisplayDashboardView.cpp]: `mpptPowerReceived` isn't being used anywhere so let's remove all references to it.

File: src/ViewLayer/DisplayDashboard/DisplayDashboardView/DisplayDashboardView.cpp [hunk 3]
```diff
@@ -296,11 +294,12 @@ void DisplayDashboardView::mpptReceived(int i, Mppt mppt)
     }
 }
 
+/*This does not appear to be used
 void DisplayDashboardView::mpptPowerReceived(double mpptPower)
```
Comment (bill-luu) [inline src/ViewLayer/DisplayDashboard/DisplayDashboardView/DisplayDashboardView.cpp]: Let's remove this too

File: src/DataLayer/KeyMotorData/I_KeyMotorData.h [hunk 4]
```diff
@@ -19,8 +19,8 @@ class I_KeyMotorData : public QObject
 signals:
     void motorZeroReceived(KeyMotor motorZero);
```
Comment (bill-luu) [inline src/DataLayer/KeyMotorData/I_KeyMotorData.h]: Let's keep consistency and have all signals be `void signal(DataType)`, without a variable name.

File: src/PresenterLayer/KeyMotorPresenter/KeyMotorPresenter.h [hunk 5]
```diff
@@ -18,7 +18,7 @@ class KeyMotorPresenter : public QObject
     void motorZeroReceived(KeyMotor motorZero);
     void motorOneReceived(KeyMotor motorOne);
     void motorSetCurrentReceived(double setCurrent);
```
Comment (bill-luu) [inline src/PresenterLayer/KeyMotorPresenter/KeyMotorPresenter.h]: As above

File: src/PresenterLayer/KeyMotorPresenter/KeyMotorPresenter.h [hunk 6]
```diff
@@ -18,7 +18,7 @@ class KeyMotorPresenter : public QObject
...
-    void motorActualSpeedReceived(double actualSpeed);
-    void motorBusVoltageReceived(double busVoltage);
+    void motorActualSpeedReceived(double);
+    void motorBusVoltageReceived(double);
     void motorBusCurrentReceived(double busVoltage);
```
Comment (bill-luu) [inline src/PresenterLayer/KeyMotorPresenter/KeyMotorPresenter.h]: As above.

**Pull request merged**